### DISCUSSION TITLE
Correct way to have a customer with multiple numbers?

### DISCUSSION THREAD

**Discussion by steve-potts** [pocketbase/pocketbase#639] [Q&A] [answered]
Hello,

So i am trying to create a customer table, with a customer_number table and also a customer_address table.
the customer could have 0 or more address and 1 or more numbers.

at first i assumed to store a relation on the customer_address and customer_number table, relating to the customer id, but then expand didnt work.

I switched that around so that the customer table has a relation to each and then expand does work, but i can only have one address and one number?

What is the correct way of achieving this?

i also prefer to use the API requests method of query rather than the client as i am using angular and i just feel it works better?
eg in angular service i have this currently. 
if its better to use client method, anyone suggest how good way to do that in angular?

`  getCustomerList() {
    return this.http.get(`${this.apiBaseUrl}?expand=customer_address_id,customer_number_id`);
  }`

thanks for any advice

**Comment by ganigeorgiev** [comment, marked as answer]
Hm, this seems similar to https://github.com/pocketbase/pocketbase/discussions/289.

> but i can only have one address and one number?

The relation field support multiple values, but you have to change the "Max select" option to be more than 1 (_in the next release you'll be able also to leave it empty for no upper limit_).

> What is the correct way of achieving this?

It depends on your use case and how you want to query the data. In the next v0.8.0 release, there will be support for indirect expand - https://github.com/pocketbase/pocketbase/issues/312, but you can also eager load your records manually by doing something similar as explained in https://github.com/pocketbase/pocketbase/discussions/289#discussioncomment-3373497.


> i also prefer to use the API requests method of query rather than the client as i am using angular and i just feel it works better?
eg in angular service i have this currently.

I'm not sure that I understand what you mean. It's up to the developer whether they'll use the SDK or construct the request on their own. The SDK should make things easier in general (eg. will add the `Authorization` header for you if you are authenticated, will normalize the `Content-Type` based on the data type, will wrap all response errors in `ClientResponseError` object, etc.).
